LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/BasicsStation/Processors/LnsProtocolMessageProcessor.cs

```diff
@@ -196,9 +196,16 @@ internal async Task InternalHandleDataAsync(RouteValueDictionary routeValues, We
                     break;
                 case var messageType and (LnsMessageType.DownlinkMessage or LnsMessageType.RouterConfig):
                     throw new NotSupportedException($"'{messageType}' is not a valid message type for this endpoint and is only valid for 'downstream' messages.");
+                case LnsMessageType.TimeSync:
+                    var timeSyncData = JsonSerializer.Deserialize<TimeSyncMessage>(json);
+                    LogReceivedMessage(this.logger, "TimeSync", json, null);
+                    timeSyncData.gpstime = (ulong)DateTime.UtcNow.Subtract(
+                        new DateTime(1980, 1, 6, 0, 0, 0, DateTimeKind.Utc) // GPS Epoch
```

Comment:
Suggest making the GpsEpoch value a static read-only property, so we don't create it multiple times.